src/csvcubed/models/validatedmodel.py

```diff
@@ -0,0 +1,42 @@
+# this class will be class attributes and returns an array of errors
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List
+
+from csvcubedmodels.dataclassbase import DataClassBase
+from csvcubed.models.validationerror import ValidateModelProperiesError
+from csvcubed.utils.validations import (
+    validate_str_type,
+    validate_int_type,
+    validate_list,
+)
+
+ValidationFunction = Callable[[Any, str], List[ValidateModelProperiesError]]
+
+
+class ValidatedModel(DataClassBase):
```

Comment:
Can you add a docstring here explaining the purpose of the `ValidatedModel` and can you make it an abstract base class too so it's not possible to directly instantiate it?

Reply:
done